csrc/ir/nodes.cpp

```diff
@@ -5756,4 +5765,177 @@ std::vector<PolymorphicValue> TopKOp::evaluate(
 
 NVFUSER_DEFINE_CLONE_AND_CREATE(TopKOp)
 
+GroupedMmaOp::GroupedMmaOp(
+    IrBuilderPasskey passkey,
+    Val* out,
+    Val* mat1,
+    Val* mat2,
+    Val* offsets,
+    Val* scale1,
+    Val* scale2)
+    : Expr(passkey) {
+  NVF_ERROR(
+      out->getValType().value() == ValType::TensorView,
+      "Output must be a TensorView");
+  NVF_ERROR(
+      mat1->getValType().value() == ValType::TensorView,
+      "First input must be a TensorView");
+  NVF_ERROR(
+      mat2->getValType().value() == ValType::TensorView,
+      "Second input must be a TensorView");
+  NVF_ERROR(
+      offsets->getValType().value() == ValType::TensorView,
+      "Offsets must be a TensorView");
+  addOutput(out);
+  addInput(mat1);
+  addInput(mat2);
+  addInput(offsets);
+
+  bool has_scale1 = scale1 != nullptr;
+  if (has_scale1) {
+    NVF_CHECK(
+        scale1->getValType().value() == ValType::TensorView,
+        "Scale1 must be a TensorView");
+    NVF_CHECK(
+        scale2->getValType().value() == ValType::TensorView,
+        "Scale2 must be a TensorView");
+    addInput(scale1);
+    addInput(scale2);
+  }
+  addDataAttribute(has_scale1);
+}
+
+std::string GroupedMmaOp::toString(int indent_size) const {
+  std::stringstream ss;
+  indent(ss, indent_size) << out()->toString() << " = GroupedMmaOp("
+                          << "mat1=" << matrix1()->toString() << ", "
+                          << "mat2=" << matrix2()->toString() << ", "
+                          << "offsets=" << offsets()->toString();
+  if (hasScale()) {
+    ss << ", "
+       << "scale1=" << scale1()->toString() << ", "
+       << "scale2=" << scale2()->toString();
+  }
+  ss << ")\n";
+  return ss.str();
+}
+
+std::string GroupedMmaOp::toInlineString(int indent_size) const {
+  NVF_CHECK(false, "Tensor op can not be printed inline");
+}
+
+std::vector<PolymorphicValue> GroupedMmaOp::evaluate(
+    const ExpressionEvaluator& ee,
+    const std::vector<PolymorphicValue>& inputs) const {
+  NVF_ERROR(
+      (inputs.size() == 3 && !hasScale()) || (inputs.size() == 5 && hasScale()),
+      "GroupedMmaOp expects 3 or 5 inputs but received ",
+      inputs.size(),
+      " with scale flag: ",
+      hasScale() ? "true" : "false");
+
+  const auto& mat1 = inputs[0];
```

Comment:
```suggestion
  NVF_ERROR(inputs[0].isA<at::Tensor>());
  const auto& mat1 = inputs[0].as<at::Tensor>();
```
because I saw the code `mat1.as<at::Tensor>()` a lot.